src/Dofs/MixedDofHandler.jl

```diff
@@ -4,30 +4,63 @@ mutable struct FieldHandler
     cellset::Set{Int}
     bc_values::Vector{BCValues} # for boundary conditions
 end
+"""
+    FieldHandler(fields::Vector{Field}, cellset)
 
+Construct a `FieldHandler` based on an array of `Field`s and assigns it a set of cells.
+"""
 function FieldHandler(fields::Vector{Field}, cellset)
     # TODO for now, only accept isoparamtric mapping
     bc_values = [BCValues(field.interpolation, field.interpolation) for field in fields]
     FieldHandler(fields, cellset, bc_values)
 end
 
+
+struct CellDofs
+    dofs::Vector{Int}
+    offset::Vector{Int}
+    length::Vector{Int}
+end
+
+import Base.getindex
+function getindex(eldofs::CellDofs, el::Int64)
+     return eldofs.dofs[eldofs.offset[el]:eldofs.offset[el]+eldofs.length[el]-1]
+ end
+
+
 struct MixedDofHandler{dim,C,T} <: JuAFEM.AbstractDofHandler
     fieldhandlers::Vector{FieldHandler}
-    cell_dofs::Vector{Int}
-    cell_dofs_offset::Vector{Int}
+    cell_dofs::CellDofs
     closed::ScalarWrapper{Bool}
     grid::MixedGrid{dim,C,T}
 end
 
 function MixedDofHandler(grid::MixedGrid)
-    MixedDofHandler(FieldHandler[], Int[], Int[], JuAFEM.ScalarWrapper(false), grid)
+    MixedDofHandler(FieldHandler[], CellDofs([],[],[]), JuAFEM.ScalarWrapper(false), grid)
 end
 
 getfieldnames(fh::FieldHandler) = [field.name for field in fh.fields]
 getfielddims(fh::FieldHandler) = [field.dim for field in fh.fields]
 getfieldinterpolations(fh::FieldHandler) = [field.interpolation for field in fh.fields]
+import JuAFEM.ndofs_per_cell, JuAFEM.celldofs, JuAFEM.isclosed, JuAFEM.ndofs
+ndofs_per_cell(dh::MixedDofHandler, cell::Int64) = dh.cell_dofs.length[cell]
+ndofs(dh::MixedDofHandler) = maximum(dh.cell_dofs.dofs)
```

Comment:
```suggestion
ndofs(dh::MixedDofHandler) = length(dh.cell_dofs.dofs)
```

If we are assuming that the celldofs start from 1 and are continous?

Reply:
They are not continuous, simply an array with all the dofs for each cell after one another.
But I can't see `ndofs` being called frequently. Only in some initializations like for the sparse structure.

Reply:
Yeah, I mixed it up with `ndofs_per_cell` -.-.

Still, I wonder if it wouldn't be better to cache it on `close!` in `dh.ndofs` or something.

Reply:
Sure, we can do that. Saving an integer may be acceptable use of memory in this case 😄